deb/index_files.go

```diff
@@ -91,11 +94,24 @@ func (file *indexFile) Finalize(signer pgp.Signer) error {
 		file.parent.generatedFiles[file.relativePath+ext] = checksumInfo
 	}
 
-	err = file.parent.publishedStorage.MkDir(filepath.Dir(filepath.Join(file.parent.basePath, file.relativePath)))
+	filedir := filepath.Dir(filepath.Join(file.parent.basePath, file.relativePath))
+
+	err = file.parent.publishedStorage.MkDir(filedir)
 	if err != nil {
 		return fmt.Errorf("unable to create dir: %s", err)
 	}
 
+	hashs := []string{}
+	if file.acquireByHash {
+		hashs = append(hashs, "MD5Sum", "SHA1", "SHA256", "SHA512")
+		for _, hash := range hashs {
```

Comment:
I think this can be straight `for _, hash := range []string{"MD5Sum", "SHA1", ....} {`

Reply:
Changed.